### PR TITLE
Use Environment.SystemDirectory to get system directory

### PR DESCRIPTION
**Pull request by jkotas**
Environment.SystemDirectory is a lot more efficient than Environment.GetFolderPath(System.Environment.SpecialFolder.System).

Contributes to https://github.com/dotnet/aspnetcore/issues/47269

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostTests.cs [hunk 1]
```diff
@@ -55,12 +55,7 @@ public void CreateDefaultBuilder_DoesNotChangeContentRootIfCurrentDirectoryIsWin
         {
             using var _ = RemoteExecutor.Invoke(() =>
             {
-                string systemDirectory = Environment.GetFolderPath(Environment.SpecialFolder.System);
-                if (string.IsNullOrEmpty(systemDirectory))
-                {
-                    // Skip the environments (like Nano Server) where Environment.SpecialFolder.System returns empty - https://github.com/dotnet/runtime/issues/21430
-                    return;
-                }
+                string systemDirectory = Environment.SystemDirectory;
```
Comment (stephentoub) [inline src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostTests.cs:58]: I assume this doesn't suffer from the same Nano issue?

**Review by stephentoub — APPROVED** [review]
We already have a code analysis rule that looks for things that should be replaced with members of Environment.  Looks like another one to add.

**Review by halter73 — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostTests.cs (hunk 1 above)
Comment (jkotas) [reply src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostTests.cs:58]: I would expect so. We have tests that verify that Environment.SystemDirectory works on Nano and Environment.SystemDirectory does not.

**Review by eerhardt — APPROVED** [review]
Lgtm. Thanks for taking care of this!

**Comment by eerhardt** [comment]
> We already have a code analysis rule that looks for things that should be replaced with members of Environment. Looks like another one to add.

I've opened https://github.com/dotnet/runtime/issues/83591 for this.

**Pull request merged**